WordPress/gutenberg
# Extract @wordpress/reusable-blocks from @wordpress/editor

## Description

This PR extracts the store part of reusable blocks logic from the editor package to a new, dedicated `@wordpress/reusable-blocks` package, and refactors it to use data layer instead of custom logic (props to @noisysocks).

The rationale is that the widgets editor would benefit from supporting reusable blocks as seen in https://github.com/WordPress/gutenberg/issues/22875. Unfortunately, a large chunk of reusable blocks logic lives in the `@wordpress/editor` package which was designed to support editing posts and would not make a good dependency for `@wordpress/edit-widgets` (and wouldn't even work). 

Solves #25853 

## How has this been tested?

Go to posts editor, and interact with reusable blocks in the following ways:

* Create a new reusable block, confirm it's visible in the inserter
* Split existing reusable block back into regular blocks
* Try "Delete from reusable blocks" toolbar menu item
* Insert reusable blocks using the inserter
* Copy/paste reusable blocks
* Undo/redo
* Anything else you can think of

When you're done, save the page and refresh the editor. Confirm all the changes were stored correctly and that your new reusable blocks are visible in the inserter.

## Types of changes

Breaking change, but only if messing with experimental features could be considered breaking.

## Checklist:
